### PR TITLE
_64gram: init at 1.1.15

### PR DESCRIPTION
**Pull request by clot27**
Add 64gram package
supersedes #294736
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by FabianRig** [comment]
Hi!

As far as I know, the name-based package directories are preferred (see https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md). 

Furthermore, your arguments should be on separate lines each.

**Comment by clot27** [comment]
@FabianRig 
image
That respective check passed tho...
I dont get the 2nd part, which arguments?

**Comment by FabianRig** [comment]
The packages which you "import" in the first line. Each package should be in a separate line. Furthermore, there shouldn't be the three dots ("..."). Best regards!

**Comment by clot27** [comment]
@FabianRig I see, I formated the .nix file through alejandra so I thought that was the desired outcome, I'll fix it. Thanks!

**Comment by FabianRig** [comment]
Hi! Since the check fails, I'd recommend using https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/README.md!

File: pkgs/by-name/_6/_64gram/package.nix [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    owner = "TDesktop-x64";
+    repo = "tdesktop";
+    rev = "v${version}";
+
+    fetchSubmodules = true;
+    hash = "sha256-3HLRv8RTyyfnjMF7w+euSOj6SbxlxOuczap5Nlizsvg=";
+  };
+
+  meta = with lib; {
+    description = "64Gram (unofficial Telegram Desktop)";
```
Comment (Aleksanaa) [inline pkgs/by-name/_6/_64gram/package.nix:21]:
```suggestion
    description = "An unofficial Telegram Desktop providing Windows 64bit build and extra features";
```

The description should make sense, but does not necessarily need to be taken from the repository description.

**Comment by clot27** [comment]
It would be great if this gets merged (if everything is fine!) as new 64gram update is approaching.

File: pkgs/by-name/_6/_64gram/package.nix [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    hash = "sha256-3HLRv8RTyyfnjMF7w+euSOj6SbxlxOuczap5Nlizsvg=";
+  };
+
+  meta = with lib; {
+    description = "An unofficial Telegram Desktop providing Windows 64bit build and extra features";
+    license = licenses.gpl3Only;
+    platforms = platforms.all;
+    homepage = "https://github.com/TDesktop-x64/tdesktop";
+    changelog = "https://github.com/TDesktop-x64/tdesktop/releases/tag/v${version}";
+    maintainers = with maintainers; [ clot27 ];
```
Comment (Aleksanaa) [inline pkgs/by-name/_6/_64gram/package.nix:26]:
```suggestion
    maintainers = with maintainers; [ clot27 ];
    mainProgram = "telegram-desktop";
```

```
=> nix run github:clot27/nixpkgs/init-64gram#_64gram
error: unable to execute '/nix/store/6b464aqwsb2aji9jl7kxq3mdbd8rll50-64Gram-1.1.15/bin/64Gram': No such file or directory
```

**Review by Aleksanaa — APPROVED** [review]
Tested and runs. Others lgtm.

File: pkgs/by-name/_6/_64gram/package.nix (hunk 2 above)
Comment (wegank) [reply pkgs/by-name/_6/_64gram/package.nix:26]: And mark as broken on darwin since it doesn't build there.

Comment (clot27) [reply pkgs/by-name/_6/_64gram/package.nix:26]: @wegank How to do that?

Comment (clot27) [reply pkgs/by-name/_6/_64gram/package.nix:26]: Done!

**Pull request merged**